front-end/src/host/HostAnnouncementQueue.ts

```diff
@@ -0,0 +1,22 @@
+import * as React from "react";
+
+export const AddAnnouncementContext = React.createContext<any>(null);
+
+export function HostAnnouncementQueue(props) {
+  const announcementAudioObjects = props.announcementAudioObjects;
+  const [currentIndex, setCurrentIndex] = React.useState(0);
+
+  React.useEffect(() => {
+    const numAnnouncements = announcementAudioObjects.length;
+    if (numAnnouncements > 0 && currentIndex < numAnnouncements) {
+      const currentAudio = announcementAudioObjects[currentIndex];
+      currentAudio.onended = () => {
+        setCurrentIndex(currentIndex+1);
```

Comment:
Trigger movement to next announcement after the current announcement is done playing. This re-renders the component with the new index, playing a new audio if it exists.